ons-spark/raw-notebooks/checkpoint-staging/r_input.R

```diff
@@ -0,0 +1,155 @@
+library(sparklyr)
+library(dplyr)
+library(DBI)
+
+sc <- sparklyr::spark_connect(
+    master = "local[2]",
+    app_name = "checkpoint",
+    config = sparklyr::spark_config())
+
+
+set.seed(42)
+new_cols <- 12
+num_rows <- 10^3
+
+start_time <- Sys.time()
+
+df = sparklyr::sdf_seq(sc, 1, num_rows) %>%
+    sparklyr::mutate(col_0 = ceiling(rand()*new_cols))
+
+for (i in 1: new_cols)
+{
+  column_name = paste0('col_', i)
+  prev_column = paste0('col_', i-1)
+  df <- df %>%
+    sparklyr::mutate(
+      !!column_name := case_when(
+        !!as.symbol(prev_column) > i ~ !!as.symbol(prev_column)))
+  
+}
+
+df %>%
+    head(10)%>%
+    sparklyr::collect()%>%
+    print()
+
+end_time <- Sys.time()
+time_taken = end_time - start_time
+
+cat("Time taken to create DataFrame", time_taken)
+
+ 
+
+sparklyr::spark_disconnect(sc)
+
+sc <- sparklyr::spark_connect(
+    master = "local[2]",
+    app_name = "checkpoint",
+    config = sparklyr::spark_config())
+
+
+config <- yaml::yaml.load_file("ons-spark/config.yaml")
+
+sparklyr::spark_set_checkpoint_dir(sc, config$checkpoint_path)
+
+ 
+start_time <- Sys.time()
+
+df1 = sparklyr::sdf_seq(sc, 1, num_rows) %>%
+    sparklyr::mutate(col_0 = ceiling(rand()*new_cols))
+
+for (i in 1: new_cols)
+{
+  column_name = paste0('col_', i)
+  prev_column = paste0('col_', i-1)
+  df1 <- df1 %>%
+    sparklyr::mutate(
+    !!column_name := case_when(
+        !!as.symbol(prev_column) > i ~ !!as.symbol(prev_column) ))
+  
+  
+  if (i %% 3 == 0) 
+  {
+    sparklyr::sdf_checkpoint(df1, eager= TRUE)
+  }
+}
+
+df1 %>%
+    head(10)%>%
+    sparklyr::collect()%>%
+    print()
+
+end_time <- Sys.time()
+time_taken = end_time - start_time
+
+
+cat("Time taken to create DataFrame: ", time_taken)
+ 
+cmd <- paste0("hdfs dfs -rm -r -skipTrash ", config$checkpoint_path)
+p <- system(command = cmd)
+
+sparklyr::spark_disconnect(sc)
+ 
+library(sparklyr)
+library(dplyr)
+
+sc <- sparklyr::spark_connect(
+    master = "local[2]",
+    app_name = "staging_tables",
+    config = sparklyr::spark_config())
+ 
+animal_rescue_csv = config$rescue_path_csv
+
+df = sparklyr::spark_read_csv(sc,
+                              path=animal_rescue_csv,
+                              header=TRUE,
+                              infer_schema=TRUE)
+
+df %>%
+    sparklyr::select( 
+             -("WardCode"), 
+             -("BoroughCode"), 
+             -("Easting_m"), 
+             -("Northing_m"), 
+             -("Easting_rounded"), 
+             -("Northing_rounded"), 
+            "EngineCount" = "PumpCount",
+            "Description" = "FinalDescription",
+            "HourlyCost" = "HourlyNotionalCostGBP",
+            "TotalCost" = "IncidentNotionalCostGBP",
+            "OriginOfCall" = "OriginofCall",
+            "JobHours" = "PumpHoursTotal",
+            "AnimalGroup" = "AnimalGroupParent") %>%
+
+    sparklyr::mutate(DateTimeOfCall = to_date(DateTimeOfCall, "dd/MM/yyyy")) %>%
+    dplyr::arrange(desc(IncidentNumber)) %>%
```

Comment:
You should probably end the pipes here as we need to make sure we're saving the output to the df variable - we only want to do head(3), collect and print just to show the dataframe but we don't want to assign the results back to the df variable.

So remove the %>% and then begin on line 127 with df %>% head(3) - that will print the top 3 rows without saving the result to the dataframe.